operator/cmd/metrics/adapter/provider/custom_metrics.go

```diff
@@ -0,0 +1,106 @@
+package provider
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/metrics/pkg/apis/custom_metrics"
+	"sigs.k8s.io/custom-metrics-apiserver/pkg/provider"
+)
+
+type CustomMetricValue struct {
+	Value  custom_metrics.MetricValue
+	Labels map[string]string
+}
+
+type CustomMetricsCache struct {
+	mtx     sync.RWMutex
+	metrics map[string]CustomMetricValue
+}
+
+// Update adds a new metricValue for the given metricName to the cache. If an item has already been present for the provided
+// metricName, the previous value will be replaced.
+func (cm *CustomMetricsCache) Update(metricName string, metricValue CustomMetricValue) {
+	cm.mtx.Lock()
```

Comment:
Just looked a bit into this and found an interesting article comparing sync.Map with the standard map (in combination with RWMutexes): https://medium.com/@deckarep/the-new-kid-in-town-gos-sync-map-de24a6bf7c2c 

Based on the arguments provided in this article I would keep the map+mutex implementation, mainly because of the missing type safety in `sync.Map`

Reply:
Thx for the nice resource! I agree 👍